homeassistant/components/flume/config_flow.py

```diff
@@ -0,0 +1,104 @@
+"""Config flow for flume integration."""
+from functools import partial
+import logging
+
+from pyflume import FlumeAuth, FlumeDeviceList
+from requests.exceptions import RequestException
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .const import BASE_TOKEN_FILENAME, CONF_CLIENT_ID, CONF_CLIENT_SECRET
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+# If flume ever implements a login page for oauth
+# we can use the oauth2 support built into Home Assistant.
+#
+# Currently they only implement the token endpoint
+#
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Required(CONF_CLIENT_ID): str,
+        vol.Required(CONF_CLIENT_SECRET): str,
```

Comment:
For OAuth2, client ID/secret need to be stored in configuration.yaml.

Reply:
Flume doesn't provide an endpoint to login, only to get a token
https://flumetech.readme.io/reference#get-tokens-1

Reply:
The reason for this is that config flows are user facing, and client ID/secret is something for the person installing it up to supply.

Reply:
That's an OAuth2 password grant request. https://tools.ietf.org/html/rfc6749#section-4.3.2

Reply:
Well I guess except for the fact that they want JSON, so not fully spec compliant 🙄

Reply:
Can you help me out with a little more color on this? I'm missing the detail on why we should add another step for the user to set this up.

Flume is giving out a unique client id and client secret to the user in their account, no developer account required so the user should have these credentials.  

https://flumetech.readme.io/docs/authentication

"This client will not be able to authenticate any users besides your own."

Reply:
Ah, it's only for the user? In that case it's ok to keep them here.